packages/server/src/models/Subcategory.ts

```diff
@@ -125,11 +125,16 @@ SubcategorySchema.statics.getSubcategoryList = async function(
 };
 
 SubcategorySchema.statics.getByStub = async function(
-  stub: string
+  stub: string,
+  includeDeletedResources: boolean = false
 ): Promise<TSubcategoryDocument | null> {
-  return await this.findOne({ stub })
+  const result = await this.findOne({ stub })
     .populate("parentCategory")
     .populate("resources");
+  if (result && !includeDeletedResources) {
+    result.resources = result.resources.filter(r => !r.deleted);
+  }
```

Comment:
Could we incorporate the filter in the query rather than mapping over all results?

Reply:
It's a good idea, but the problem is the the query is for the subcategory. The subcategory already has references to the deleted resources (and the non-deleted ones).

Reply:
As a side note, the whole populate thing is problematic since, depending on whether it's called, the Subcategory Document could be an array of resource IDs or an array of resource documents. I'm thinking about doing a refactor so we never call `populate` and then just do all the ID -> Type conversions at the very end.